### PR TITLE
Add Implementation for adl_handler Functions

### PR DESCRIPTION
**Pull request by ametel01**
## Pull Request Type
- Feature

Closes: #275

## Breaking Change?
No

## Additional Details
Temporarily made `fn get_execute_order_params()` public in `base_order_handler.cairo`. Open to alternatives for closer alignment with Solidity.

**EDIT**
The issue mentioned in the additional details has been resolved.

### CODE REVIEW

File: src/exchange/adl_handler.cairo [hunk 1]
```diff
@@ -145,9 +163,17 @@ mod AdlHandler {
             swap_handler_address,
             referral_storage_address
         );
+        self.data_store.write(IDataStoreSafeDispatcher { contract_address: data_store_address });
```
Comment (Sk8erboi84) [inline src/exchange/adl_handler.cairo:166]: Cant we just access this fields from IBaseOrderHandler?

File: src/exchange/adl_handler.cairo [hunk 2]
```diff
@@ -145,9 +163,17 @@ mod AdlHandler {
...
+        self
+            .event_emitter
+            .write(IEventEmitterSafeDispatcher { contract_address: event_emitter_address });
+        self.oracle.write(IOracleSafeDispatcher { contract_address: oracle_address });
+        self.chain.write(IChainSafeDispatcher { contract_address: chain_address });
```
Comment (Sk8erboi84) [inline src/exchange/adl_handler.cairo:171]: Why did you add chain here ?

File: src/exchange/adl_handler.cairo [hunk 3]
```diff
@@ -108,7 +117,14 @@ mod AdlHandler {
     //                              STORAGE
     // *************************************************************************
     #[storage]
-    struct Storage {}
+    struct Storage {
+        data_store: IDataStoreSafeDispatcher,
+        event_emitter: IEventEmitterSafeDispatcher,
+        oracle: IOracleSafeDispatcher,
+        chain: IChainSafeDispatcher,
```
Comment (Sk8erboi84) [inline src/exchange/adl_handler.cairo:124]: why chain?

**Comment by Sk8erboi84** [comment]
Can you please implement tests.

File: src/exchange/adl_handler.cairo (hunk 2 above)
Comment (ametel01) [reply src/exchange/adl_handler.cairo:171]: Removed leftover code that originated from a previous closed pull request.

File: src/exchange/adl_handler.cairo (hunk 3 above)
Comment (ametel01) [reply src/exchange/adl_handler.cairo:124]: Removed leftover code that originated from a previous closed pull request.

File: src/exchange/adl_handler.cairo (hunk 1 above)
Comment (ametel01) [reply src/exchange/adl_handler.cairo:166]: Could you clarify what you're asking?

Comment (zarboq) [reply src/exchange/adl_handler.cairo:166]: image

I think something like that could be possible, since data_store is already stored in BaseOrderHandler.

Here data_store and others are stored both in BaseOrderHandler and AdlHandler

Comment (ametel01) [reply src/exchange/adl_handler.cairo:166]: Given the changes, can we consider removing all constructor parameters from `adl_handler`?

Comment (ametel01) [reply src/exchange/adl_handler.cairo]: Updated the code as requested.

**Review by abdelhamidbakhta — APPROVED** [review]
Lgtm
